jujugui/static/gui/src/app/app.js

```diff
@@ -487,13 +482,12 @@ YUI.add('juju-gui', function(Y) {
           macaroons = credentials.macaroons;
         }
       }
-      env.setCredentials({
-        user: user,
-        password: password,
-        macaroons: macaroons
-      });
+      env.setCredentials({ user, password, macaroons });
       this.env = env;
-
+      // If we're in legacy Juju (Juju 1) then we'll only require
+      // a single websocket connection.
```

Comment:
Also, this comment would make sense if we checked if (controllerAPI) here rather than inside the setUpControllerAPI method. Which I think we should do to make the flow easier to read.